storage/gcsutil/gcsobject.go

```diff
@@ -0,0 +1,302 @@
+package gcsutil
+
+import (
+	"google.golang.org/api/googleapi"
+	"google.golang.org/api/storage/v1"
+)
+
+//Changes made:
+//1) made Size in Object pointer so omitempty wont remove 0
+//2) made ObjectSize in RewriteResponse pointer so omitempty wont remove 0
+//3) made TotalBytesRewritten in RewriteResponse pointer so omitempty wont remove 0
```

Comment:
So this is a clone of the internal storage.Object, but with these changes so that JSON serialization does what you want?

Instead of making these pointers, couldn't you just remove the `omitempty`?

Reply:
its a clone of google go storage.Object. can't edit in it since it is a library

Reply:
size should always return null is only returned if specified by the caller  not to include size in returned fields otherwise if expliclity specified or by default it should be returned as 0 meas the file is empty which is a valid value. the definition of Size (primitive types) itself is wrong it shouldn't be int64 it should be a pointer instead so that null would only omit the value while 0 don't

Reply:
Hmm, but this is the Google-provided client API... are you sure this behavior is demanded by specification and not simply an implementation detail of how it happens to work?

Does this discrepancy break existing Google-provided GCS client libraries?  I'm curious what kind of failure you've run into that's motivating this change.